### PR TITLE
EDU-232 Add lessons collection

### PR DESCRIPTION
**Pull request by laurentiu-ilici**
Closes: https://educandu.atlassian.net/browse/EDU-232

### CODE REVIEW

File: src/domain/schemas/document-schemas.js [hunk 1]
```diff
@@ -53,7 +53,7 @@ export const hardDeleteDocumentBodySchema = joi.object({
   documentKey: idOrKeySchema.required()
 });
 
-const sectionDBSchema = joi.object({
+export const sectionDBSchema = joi.object({
```
Comment (laurentiu-ilici) [inline src/domain/schemas/document-schemas.js:56]: We probably should not use this, but for now it's okay. It has some properties which I think are debatable if we need such as "deleteBecause"

File: src/domain/schemas/lesson-schemas.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import joi from 'joi';
+import { idOrKeySchema } from './shared-schemas.js';
+import { slugSchema, sectionDBSchema } from './document-schemas.js';
```
Comment (ahelmberger) [inline src/domain/schemas/lesson-schemas.js:3]: We should move those two schemas into `shared-schemas.js` as they are used in multiple places

File: src/domain/schemas/document-schemas.js (hunk 1 above)
Comment (ahelmberger) [reply src/domain/schemas/document-schemas.js:56]: True, that's a good question. See also my other comment regarding this.

File: src/domain/schemas/lesson-schemas.js (hunk 2 above)
Comment (laurentiu-ilici) [reply src/domain/schemas/lesson-schemas.js]: Done, thanks!

Comment (ahelmberger) [reply src/domain/schemas/lesson-schemas.js]: When you move the `slugSchema` from `document-schemas.js` to `shared-schemas.js`, you also have to move the tests for `slugSchema` from `document-schemas.spec.js` to `shared-schemas.spec.js`.

Comment (laurentiu-ilici) [reply src/domain/schemas/lesson-schemas.js]: True, moved and added some section schema db tests.

Comment (ahelmberger) [reply src/domain/schemas/lesson-schemas.js]: Cool, thanks

**Review by ahelmberger — APPROVED** [review]

**Pull request merged**